python/nav/models/profiles.py

```diff
@@ -406,6 +406,24 @@ class Meta(object):
     def __str__(self):
         return self.type.scheme() + self.address
 
+    def has_valid_address(self):
+        from nav.alertengine.dispatchers.email_dispatcher import Email
+        from nav.alertengine.dispatchers.slack_dispatcher import Slack
+        from nav.alertengine.dispatchers.sms_dispatcher import Sms
+
+        if not self.type.supported:
+            return False
+        elif self.type.handler == 'sms':
+            if not Sms.is_valid_address(self.address):
+                return False
+        elif self.type.handler == 'email':
+            if not Email.is_valid_address(self.address):
+                return False
+        elif self.type.handler == 'slack':
+            if not Slack.is_valid_address(self.address):
+                return False
+        return True
+
```

Comment:
I don't particularly care for this if/else-structure.  Every time we add a new dispatcher, we would have to update this logic structure as well.

`AlertAddress.type` is already a relation to an `AlertSender`.  The `AlertSender` has its own logic to find and load its corresponding handler class in `AlertSender._load_dispatcher_class()`.  IMHO, a cleaner solution would be to change that into to a public method (i.e. no leading `_`), and if it does return a class, use the `is_valid_address()` call on that return value.